ompi/mca/coll/base/coll_base_allreduce.c

```diff
@@ -1247,133 +1247,109 @@ int ompi_coll_base_allreduce_intra_redscat_allgather(
     return err;
 }
 
-/**
- * A greedy algorithm to exchange data among processes in the communicator via
- * an allgather pattern, followed by a local reduction on each process. This
- * avoids the round trip in a rooted communication pattern, e.g. reduce on the
- * root and then broadcast to peers.
+/*
+ *   ompi_coll_base_allreduce_intra_allgather_reduce
  *
- * This algorithm supports both commutative and non-commutative MPI operations.
- * For non-commutative operations the reduction is applied to the data in the
- * same rank order, e.g. rank 0, rank 1, ... rank N, on each process.
+ *   Function:       use allgather for allreduce operation (for internode)
+ *   Accepts:        Same as MPI_Allreduce()
+ *   Returns:        MPI_SUCCESS or error code
  *
- * This algorithm benefits inter-node allreduce over a high-latency network.
- * Caution is needed on larger communicators(n) and data sizes(m), which will
- * result in m*n^2 total traffic and potential network congestion.
+ *   Description:    Implements allgather based allreduce aimed to improve internode 
+ *                   allreduce latency: this method takes advantage of the send and 
+ *                   receive can happen at the same time; first step is allgather
+ *                   operation to allow all ranks to obtain the full dataset; the second
+ *                   step is to do reduction on all ranks to get allreduce result. 
+ *
+ *   Limitations:    This method is designed for small message sizes allreduce because it 
+ *                   is not efficient in terms of network bandwidth comparing
+ *                   to gather/reduce/bcast type of approach; it should be combined with
+ *                   allgather bruck's method for larger communicator size.
```

Comment:
I doubt if we need to mention bruck allgather here - I guess the choice also depends on whether the communication is intra-node only or inter-node(the network characteristic also matters).